src/ng/timeout.js

```diff
@@ -91,6 +91,28 @@ function $TimeoutProvider() {
       }
       return false;
     };
+    
+    
+    /**
+    * @ngdoc method
+    * @name $timeout#cancelAll
+    *
+    * @description
+    * Cancels all timeouts that are still pending
+    *
+    * @returns {Object} Returns count of successful and failed cancelled timeouts
+    */
+    timeout.cancelAll = function() {
+      var errorOccurred = false;
+      for (var $$timeoutId in deferreds) {
+        try {
+          timeout.cancel(deferreds[$$timeoutId].promise);
```

Comment:
doing a timeout cancel rejects the original promise. How do we make sure that we won't run any extra code and get into a bad situation as a result? both for the current codebase and the future code that could get introduced?

Reply:
Talked with Eliran and Deepak offline, this work will fall under Jeremie's work with cancel all promises.

Reply:
Talked with Eliran and Deepak offline, this work will fall under Jeremie's work with cancel all promises.